### ISSUE TITLE
Release with javascript build

### ISSUE THREAD

**Issue by charlike** [PolymerLabs/lit-html#70]
It's completely unusable and inaccessible currently, or i'm an idiot :D In package.json don't have `files` field and only expose `main` that is to not existing file.

Currently i clone, compile with TS and then build cjs with rollup (because i'm trying it using `budo` server so it need cjs). But in anyway, the TS compiled `5` to some strange thing `NodeFilter.SHOW_TEXT` and everything fails. That okey, i debugged everything and replaced this with `5` because in the source code is `5` ([here](https://github.com/PolymerLabs/lit-html/blob/master/src/lit-html.ts#L113), and [here](https://github.com/PolymerLabs/lit-html/blob/master/src/lit-html.ts#L436))

That `5 /* elements & text */` was compiled to `NodeFilter.SHOW_ELEMENT | NodeFilter.SHOW_TEXT`)

But when trying to build `lit-extended` Rollup fails with `magicString of undefined` error, which is error that was happening before updating the mentioned above.

**Comment by jimsimon** [comment]
@charlike Depending on what you need there's an older version published on NPM: https://www.npmjs.com/package/lit-html

The only feature I noticed that was missing was the unsafeHTML directive, which is pretty easy to recreate yourself.

**Comment by justinfagnani** [comment]
@charlike lit-html is distributed as JS modules so that they load on browsers with native module support. Like all the other Polymer modules on npm, it's intended to be compiled at the application level, because applications know what their target environment supports.

I can definitely fix `main` and `files`, and look at Rollup. I'm not sure how best to distribute an ES5 build.

**Comment by justinfagnani** [comment]
@charlike I just tried rollup, and didn't see any error. Can you give a reproduction?

**Comment by jimsimon** [comment]
I misread this the first time.  I know Babel handles the published lit-html package without error for me.  One thing that seems odd here is that there's mention of cloning, so it seems like @charlike is building from source. 
Is it possible that what was on the master branch was broken for a bit?  It's also not clear that any transpilation step is being done after the typescript compiler is run, so that could cause issues in every browser except Safari.

**Comment by charlike** [comment]
No no. The whole point was that when i installed it, it doesn't have compiled (from ts) source to js. So, it was inaccessible through anything, because these files that are pointed from `main` and `module` fields wasn't there.

When i done clear install now, it appears that everything is okey. But again, when looking on the compiled js files, I still see `NodeFilter.SHOW_ELEMENT | NodeFilter.SHOW_TEXT` in two places, which will break the things. It's pretty weird why TypeScript compiles number `5` to this..

***

> I can definitely fix `main` and `files`, and look at Rollup. I'm not sure how best to distribute an ES5 build.

You can pretty easy use rollup to compile TS (seems that [rollup-plugin-typescript2](https://github.com/ezolenko/rollup-plugin-typescript2) is the way to go) to JS and make all needed bundles - ES, CJS, UMD, so anyone will be happy, even es5 guys.

**Comment by charlike** [comment]
Probably such config.

```js
const { minify } = require('uglify-es')
const uglify = require('rollup-plugin-uglify')
const gzip = require('rollup-plugin-gzip')
const buble = require('rollup-plugin-buble')
const typescript2 = require('rollup-plugin-typescript2')

let config = {}

if (process.env.BROWSER) {
  config = {
    input: `src/lit-html.js`,
    name: 'litHtml',
    sourcemap: true,
    plugins: [
      typescript2(),
      // seems better than using Babel,
      // gives better and smaller results.
      buble({
        target: {
          ie: '10',
          edge: '12',
          safari: '9', // latest available in buble ;/
          chrome: '52', // latest available in buble ;/
          firefox: '48', // latest available in buble ;/
        },
      }),
      uglify({ compress: { warnings: false } }, minify),
      gzip(),
    ],
    output: { file: `dest/lit-html.umd.js`, format: 'umd' },
  }
} else {
  config = {
    input: `src/lit-html.js`,
    // no need for buble, if you don't need
    // too target older node
    plugins: [typescript2()],
    output: [
      { file: `dest/lit-html.js`, format: 'cjs' },
      { file: `dest/lit-html.es.js`, format: 'es' },
    ],
  }
}

module.exports = config
```

use it:

```
$ BROWSER=true rollup -c
$ rollup -c
```

**Comment by justinfagnani** [comment]
@charlike you're saying you didn't have a top-level `lit-html.js`?

**Comment by charlike** [comment]
Yep, it was strange, that's why i opened the issue. I probably should close that issue and open another for the `NodeFilter.SHOW_ELEMENT | NodeFilter.SHOW_TEXT` thing.

**Comment by jimsimon** [comment]
I guess I'm not following the part about needing to compile the typescript files.  I installed the package from npm via `npm install lit-html` and it works fine for me.  If you're installing directly from GitHub (which it sounds like you are/we're), then you'll definitely need to compile manually since the built assets aren't commited to the repo (and really shouldn't be anyway).

**Comment by charlike** [comment]
@jimsimon i fallback to install from github, _because_ the npm package wasn't including them. Anyway, it is working today, but there's another issue.

**Comment by daflair** [comment]
By the way… I've been also having trouble getting lit-html to build the dest when added as a dependency. I use yarn add for local development and npm install on travis.

I have not been keeping good records on this issue yet but I know that I did try to install it as "lit-html" and "PolymerLabs/lit-html" at different times over the past few days, and opt-ed for the later with the rushed assumption that the repo might be the more reliable source. I cannot confirm that either one actually had the built dest folder yet, but now I have time to start looking into it.

In my project I had to resort to manually building it in my ci script and that is extremely fragile but I'm only at alpha.2.

Here is what I'm working on: [polyestr-tsx-lit-transform](https://github.com/grasppe/polyestr-tsx-lit-transform) ([alpha.2](https://github.com/grasppe/polyestr-tsx-lit-transform/tree/1.0.0-alpha.2)).

Now that I found this thread, I can really use your advise on the best way to add lit-html as a dependency and as an optionalDependency. How should it be defined in my package.json. Also, how should it be installed with yarn add, and npm install to get the most current version and have it automatically include the built files.
